src/util/ChartUtils.ts

```diff
@@ -17,10 +17,19 @@ import { findEntryInArray, getPercentValue, isNumber, isNumOrStr, mathSign, uniq
 import { filterProps, findAllByType, findChildByType, getDisplayName } from './ReactUtils';
 // TODO: Cause of circular dependency. Needs refactor.
 // import { RadiusAxisProps, AngleAxisProps } from '../polar/types';
-import { AxisType, BaseAxisProps, DataKey, LayoutType, LegendType, PolarLayoutType, TickItem } from './types';
+import {
+  AxisType,
+  BaseAxisProps,
+  DataKey,
+  LayoutType,
+  LegendType,
+  PolarLayoutType,
+  SpecificDomain,
+  TickItem,
+} from './types';
 import { Payload as LegendPayload } from '../component/DefaultLegendContent';
 
-export function getValueByDataKey<T>(obj: T, dataKey: DataKey<any>, defaultValue?: any) {
+export function getValueByDataKey<T>(obj: T, dataKey: DataKey<T>, defaultValue?: any) {
```

Comment:
Strictly speaking this could be a breaking change, since it is exported. But I would rather consider this a fix, an object with an unrelated dataKey makes no sense.

Reply:
As far as I can see the DataKey is only ever used as `DataKey<any>` anyway. But yes could be breaking.